src/queries/home.js

```diff
@@ -0,0 +1,41 @@
+import { useStaticQuery, graphql } from "gatsby"
+
+export const useHomePageData = () => {
+  const {allMarkdownRemark} = useStaticQuery (
+    graphql`
+      query HomePageData {
+        allMarkdownRemark(filter: {frontmatter: {template: {eq: "home"}}}) {
+          edges {
+            node {
+              id
+              frontmatter {
+                headerSecton {
+                  mainText
+                  subText
+                  buttonText
+                  buttonLink
+                  image
+                }
+                aboutSection {
+                  title
+                  mainText
+                  mediumUrl
+                }
+                projectsSection {
+                  title
+                  projects {
+                    title
+                    description
+                    image
+                    slug
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    `
+  )
+  return allMarkdownRemark.edges[0].node.frontmatter
```

Comment:
are we sure this `edges` array will definitely have an element in it? Unless better to handle. You might end up with calling attributes on an undefined object. How about using something like `Lodash` for this?

Reply:
If there is data in the MD file, the edges array will definitely have an element in it! But as you suggested better to add a validation check!